src/extractor/guidance/intersection_generator.cpp

```diff
@@ -277,7 +277,7 @@ Intersection IntersectionGenerator::mergeSegregatedRoads(Intersection intersecti
     {
         const double correction_factor = (intersection[1].turn.angle) / 2;
         for (std::size_t i = 2; i < intersection.size(); ++i)
-            intersection[i].turn.angle += correction_factor;
+            intersection[i].turn.angle -= correction_factor;
```

Comment:
Can you add a comment outlining why the angle needs to be adjusted? It is not obvious to me from the surrounding code. Similarly why do you subtract here but add above?